venv/pysnes/cpu.py

```diff
@@ -512,20 +512,170 @@ def fetch_decode_execute(self, code):
             self.PC = self.PC + 1
         # LDX #const
         elif opcode == 0xA2:
-            const = self.fetch_byte(code)
-            result = const
-            self.compute_flags(result, self.isM())
-            self.X = result
-            self.cycles += 2
-            self.PC = self.PC + 1
+            if self.isX(): # 8 Bit Y/X
+                const = self.fetch_byte(code) # X=1 -> 8 Bit X -> one byte
+                result = const
+                self.compute_flags(result, self.isX())
+                self.X = result
+                self.cycles += 3 - self.x()
+                self.PC = self.PC + 1
+            else: # 16 Bit X/Y
+                const = self.fetch_twobyte(code) # X=0 -> 16 Bit X -> two byte
+                result = const
+                self.compute_flags(result, self.isX())
+                self.X = result
+                self.cycles += 3 - self.x()
+                self.PC = self.PC + 1
+        # LDX dp
+        elif opcode == 0xA6:
+            if self.isX(): # 8 Bit X/Y
+                addr = self.fetch_byte(code)
+                wrapped_addr = (addr + self.DP) & 0xFFFF # direct page wrapping
```

Comment:
again, please clarify what & 0xFFFF is supposed to do

Reply:
0xFFFF does direct page wrapping. Using the dp register to compute an adress, the adress ist wrapped. For example DP=0xFFFE and addr = 0x2 DP+addr=0X0001 and *NOT* 0x10001!

Reply:
Again, make sure to use 0xFFFE and not 0xFFFF in the code.